src/test/util_expected_tests.cpp

```diff
@@ -17,25 +22,39 @@ BOOST_AUTO_TEST_CASE(expected_value)
         int x;
     };
     Expected<Obj, int> e{};
-    BOOST_CHECK(e.value().x == 0);
+    BOOST_CHECK_EQUAL(e.value().x, 0);
 
     e = Obj{42};
 
     BOOST_CHECK(e.has_value());
```

Comment:
nit: Should this be a require since otherwise we would start triggering internal asserts or exceptions further down on failure?
```suggestion
    BOOST_REQUIRE(e.has_value());
```

Reply:
I think it's slightly better as a general principle, to make test failures more intelligible. But it's very unlikely that anyone would even locally tweak the code being tested here so that it fails.